src/java.desktop/windows/native/libawt/windows/awt_TrayIcon.cpp

```diff
@@ -221,6 +223,22 @@ void AwtTrayIcon::InitNID(UINT uID)
     m_nid.uVersion = NOTIFYICON_VERSION;
 }
 
+// Call updateImage() method on the peer when screen scale changes
+void AwtTrayIcon::UpdateImageDPI()
+{
+    JNIEnv *env =(JNIEnv *)JNU_GetEnv(jvm, JNI_VERSION_1_2);
+
+    jobject peer = GetPeer(env);
+    if (peer != NULL) {
+        env->CallVoidMethod(peer, idUpdateImage);
+    }
+
+    if (safe_ExceptionOccurred(env)) {
+        env->ExceptionDescribe();
+        env->ExceptionClear();
+    }
```

Comment:
Reduce this to ExceptionClear()
(1) ExceptionDescribe() per the docs https://docs.oracle.com/en/java/javase/17/docs/specs/jni/functions.html#exceptions
prints a message .. which we don't want .. it is meant for debugging not production and FWIW it clears the pending exception so any time you see Describe() followed by Clear() the code seems odd (well pointless)
2) safe_ExceptionOccurred is only useful if you are coming from a JNI method that uses C++ exception handling. Here the call originates from native doesn't it ? And we don't return to Java since it is a native event loop.

So JUST env->ExceptionClear() will do

Reply:
@prrace Thanks for the details on ExceptionClear and its usage. Makes sense. I'll have it fixed on my next commit.